apps/web/src/lib/variable-models/managed/asset-additional-info-key-variable-model.ts

```diff
@@ -34,36 +34,23 @@ export default class AssetAdditionalInfoKeyVariableModel implements IBaseVariabl
 
             if (!this.#fetcher) this.#fetcher = getCancellableFetcher(SpaceConnector.clientV2.inventory.cloudServiceQuerySet.list);
 
-            const _query: Record<string, any> = {
-                only: ['additional_info_keys'],
-                filter: [
-                    {
-                        key: 'additional_info_keys',
-                        value: null,
-                        operator: 'not',
-                    },
-                ],
-            };
-            if (query.filters?.length) {
-                apiQueryHelper.setFilters([{ k: 'additional_info_keys', v: query.filters, o: '=' }]);
-                _query.filter.push(...(apiQueryHelper.data?.filter ?? []));
-            }
-            if (query.search) {
-                _query.filter.push({
-                    key: 'additional_info_keys',
-                    value: query.search,
-                    operator: 'contain',
-                });
-            }
+            apiQueryHelper.setOnly('additional_info_keys')
+                .setFilters([{
+                    k: 'additional_info_keys',
+                    v: null,
+                    o: '!=',
+                }]);
             const { status, response } = await this.#fetcher({
                 ...dependencyOptions,
-                query: _query,
+                query: apiQueryHelper.data,
             });
             if (status === 'succeed' && response.results?.length) {
                 const target = response.results[0]?.additional_info_keys ?? [];
-                this.#response = {
-                    results: target.map((d) => ({ key: `additional_info.${d}`, name: d })),
-                };
+                let _results = target.map((d) => ({ key: `additional_info.${d}`, name: d }));
+                if (query.search) {
+                    _results = _results.filter((d) => d.name.toLowerCase().includes(query.search as string));
```

Comment:
suggestion: I think it would be better to use getHighlightText() like other autocomplete features!